integration-cli/docker_cli_events_test.go

```diff
@@ -329,85 +329,66 @@ func TestEventsFilterImageName(t *testing.T) {
 	logDone("events - filters using image")
 }
 
-func TestEventsFilterContainerID(t *testing.T) {
-	since := daemonTime(t).Unix()
+func TestEventsFilterContainer(t *testing.T) {
 	defer deleteAllContainers()
+	since := fmt.Sprintf("%d", daemonTime(t).Unix())
+	nameID := make(map[string]string)
 
-	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "-d", "busybox", "true"))
-	if err != nil {
-		t.Fatal(out, err)
-	}
-	container1 := strings.TrimSpace(out)
-
-	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "run", "-d", "busybox", "true"))
-	if err != nil {
-		t.Fatal(out, err)
-	}
-	container2 := strings.TrimSpace(out)
-
-	for _, s := range []string{container1, container2, container1[:12], container2[:12]} {
-		if err := waitInspect(s, "{{.State.Running}}", "false", 5); err != nil {
-			t.Fatalf("Failed to get container %s state, error: %s", s, err)
-		}
-
-		eventsCmd := exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", daemonTime(t).Unix()), "--filter", fmt.Sprintf("container=%s", s))
-		out, _, err := runCommandWithOutput(eventsCmd)
+	for _, name := range []string{"container_1", "container_2"} {
+		out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "true"))
 		if err != nil {
-			t.Fatalf("Failed to get events, error: %s(%s)", err, out)
+			t.Fatal(err)
 		}
-		events := strings.Split(out, "\n")
-		checkEvents(t, events[:len(events)-1])
+		nameID[name] = strings.TrimSpace(out)
+		waitInspect(name, "{{.State.Runing }}", "false", 5)
 	}
 
-	logDone("events - filters using container id")
-}
+	until := fmt.Sprintf("%d", daemonTime(t).Unix())
 
-func TestEventsFilterContainerName(t *testing.T) {
-	since := daemonTime(t).Unix()
-	defer deleteAllContainers()
+	checkEvents := func(id string, events []string) error {
+		if len(events) != 3 { // create, start, die
+			return fmt.Errorf("expected 3 events, got %v", events)
+		}
+		for _, event := range events {
+			e := strings.Fields(event)
+			if len(e) < 3 {
+				return fmt.Errorf("got malformed event: %s", event)
+			}
 
-	_, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "--name", "container_1", "busybox", "true"))
-	if err != nil {
-		t.Fatal(err)
+			// Check the id
+			parsedID := strings.TrimSuffix(e[1], ":")
+			if parsedID != id {
+				return fmt.Errorf("expected event for container id %s:\n%s", id, event)
```

Comment:
I just like having this info on a clean line, I suppose since it is just one event it's no big deal. I think I actually intended this to show all the events :)